pkg/useragent/analyzer.go

```diff
@@ -124,22 +124,16 @@ func (a *analyzer) start() {
 
 					newRecordFlag := false // have we created a rare signature entry for this host in this chunk yet?
 
-					type hostEntry struct {
-						CID int `bson:"cid"`
-					}
-
-					var hostEntries []hostEntry
-
 					entryHostQuery := rareSigIP.BSONKey()
 					entryHostQuery["dat.rsig"] = datum.Name
-					//TODO: Consider adding the chunk to the query instead of checking it after the query
+					entryHostQuery["cid"] = a.chunk
 
-					_ = ssn.DB(a.db.GetSelectedDB()).C(a.conf.T.Structure.HostTable).Find(entryHostQuery).All(&hostEntries)
+					nExistingEntries, _ := ssn.DB(a.db.GetSelectedDB()).C(a.conf.T.Structure.HostTable).Find(entryHostQuery).Count()
 
 					//TODO: I think there might be a bug here. From what I can tell, if we set the new record flag
 					// when there is a rare signature record from a previous chunk, then we will end up pushing a duplicate
 					// record for each chunk. We should investigate removing this chunk check here. -LL
-					if len(hostEntries) <= 0 || hostEntries[0].CID != a.chunk {
```

Comment:
Moved the chunk check into the mongoDB query ( the first TODO). However, this analyzer is pretty much completely rewritten in #683 and there will be merge conflicts.